File: smoke/smoke.py
import json
import c5_envelope as c5
from c5_envelope.lang.python.envelope import t1_to_dict
from c5_envelope.simple_envelope import SimpleEnvelopeProps

pay: c5.PayloadT = c5.PayloadT.from_dict({
  'kind': "kind",
  'data': { 'y': 4 },
})

env: c5.EnvelopeT = c5.EnvelopeT.from_dict({
    'v': "A",
    'id': "id",
    'src': "envelope",
    'dst': [],
    't': 4711,
    'ttl': 10,
    'data': pay.to_dict(),
})

message = c5.SimpleEnvelopeProps(**{
  'src': "test case",
  'data': pay.to_dict()
})

refStr = c5.simpleEnvelope(message).asJson()
ref = c5.EnvelopeT.from_dict(json.loads(refStr))
env= c5.simpleEnvelope(c5.SimpleEnvelopeProps(**(ref.to_dict())))
envStr = env.asJson()
if (refStr != envStr):
  raise Exception(f'ref={refStr} env={envStr}')

print("Ready for production")

File: smoke/smoke.ts
import * as c5 from "c5-envelope";

const pay: c5.Payload<{ y: number }> = {
  kind: "kind",
  data: { y: 4 },
};

const env: c5.Envelope<{ y: number }> = {
  v: "A",
  id: "id",
  src: "envelope",
  dst: [],
  t: 4711,
  ttl: 10,
  data: pay,
};

const payt: c5.PayloadT = c5.Convert.toPayload(
  JSON.stringify({
    kind: "kind",
    data: { y: 4 },
  })
);

const envt: c5.EnvelopeT = c5.Convert.toEnvelope(
  JSON.stringify({
    v: "A",
    id: "id",
    src: "envelope",
    dst: [],
    t: 4711,
    ttl: 10,
    data: payt,
  })
);

const message: c5.SimpleEnvelopeProps = {
  src: "test case",
  data: payt,
};

const ref = c5.Convert.toEnvelope(c5.simpleEnvelope(message).asJson());
const refStr = JSON.stringify(ref);
const envStr = c5.simpleEnvelope(c5.simpleEnvelope(ref).asEnvelope()).asJson();
if (refStr !== envStr) {
  throw Error(`ref=${refStr} env=${envStr}`);
}
console.log("Ready for production");
